pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -38,8 +38,9 @@ let
       # Necessary for BTF
       DEBUG_INFO                = mkMerge [
         (whenOlder "5.2" (if (features.debug or false) then yes else no))
-        (whenAtLeast "5.2" yes)
+        (whenBetween "5.2" "5.18" yes)
```

Comment:
I dug up https://salsa.debian.org/kernel-team/linux/-/commit/6834606ba61e797a3bfd88731a0e8924b1566170 which means we only need to set DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT to y on 5.18.

Reply:
Isn't that what I'm doing here?  `whenBetween "5.2" "5.18"` means including 5.2, but not including 5.18.  So on 5.18 we only set DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT, not DEBUG_INFO.

Reply:
Oh, yeah, somehow missed that.